cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -219,31 +219,24 @@ def download_chunk(
     def download_frames(
         self,
         frame_ids: Sequence[int],
+        image_extension: str,
         *,
-        outdir: StrPath = ".",
+        outdir: str = ".",
         quality: str = "original",
         filename_pattern: str = "frame_{frame_id:06d}{frame_ext}",
     ) -> Optional[List[Image.Image]]:
         """
-        Download the requested frame numbers for a task and save images as outdir/filename_pattern
+        Download the requested frame numbers for a job and save images as outdir/filename_pattern
         """
-        # TODO: add arg descriptions in schema
 
         outdir = Path(outdir)
-        outdir.mkdir(exist_ok=True)
+        outdir.mkdir(parents=True, exist_ok=True)
 
         for frame_id in frame_ids:
             frame_bytes = self.get_frame(frame_id, quality=quality)
 
             im = Image.open(frame_bytes)
-            mime_type = im.get_format_mimetype() or "image/jpg"
-            im_ext = mimetypes.guess_extension(mime_type)
-
-            # FIXME It is better to use meta information from the server
-            # to determine the extension
-            # replace '.jpe' or '.jpeg' with a more used '.jpg'
-            if im_ext in (".jpe", ".jpeg", None):
-                im_ext = ".jpg"
+            im_ext = f".{image_extension.strip('.')}"
```

Comment:
Why the default extension logic has been removed? Changing extension should be optional.

Reply:
I thought, instead of inferring the extension from the MIME type, we can directly use the provided extension

Reply:
Yes, but only if it's provided. Tasks can include images with different extensions, so you need to get this information from the file by default.

Reply:
Oh okay. 

> Changing extension should be optional

I get this point now. I will make changes

Reply:
Yeah. So you mean that if no extension is provided then I should infer it from the MIME type, else go ahead with the provided extension ?

Reply:
Yes, everything should be as it is now if no specific extension is requested.